src/navigation/state.py

```diff
@@ -71,3 +75,29 @@ def evaluate(self, ud):
         cmd_vel = Twist()
         self.context.rover.send_drive_command(cmd_vel)
         return DoneStateTransitions.idle.name  # type: ignore
+
+
+class OffStateTransitions(Enum):
+    _settings_ = NoAlias
+
+    idle = "OffState"
+    begin_course = "WaypointState"
+
+
+class OffState(BaseState):
+    def __init__(self, context: Context):
+        super().__init__(
+            context,
+            add_outcomes=[transition.name for transition in OffStateTransitions],  # type: ignore
+        )
+
+    def evaluate(self, ud):
+        # Check if we need to ignore on
+        if self.context.course and (not self.context.course.is_complete()):
```

Comment:
perhaps I'm just missing something, but it seems like if the StateMachine had a course before recieving an Off signal, this check would cause it to immediately resume the course upon entering the off state since the course is never invalidated upon entering OffState.

Reply:
it is invalidated right before we switch, if you look in BaseState and then the transition to OffState